src/vehicle.h

```diff
@@ -239,7 +239,7 @@ struct vehicle_part : public JsonSerializer, public JsonDeserializer
     int hp() const;
 
     /** parts are considered broken at zero health */
-    bool is_broken() const;
+    bool is_broken() const { return base.damage() >= base.max_damage(); }
```

Comment:
> The whole invocation of damage here could be dropped by implementing item.is_broken()

Yes, that would work but I'm not sure if it would be significantly faster and it does push code that really is `vehicle` log into `item`. We should perhaps aim next instead to find out why we need to call `is_broken()` so much.

> floor may be fragile to rounding errors on save/load, as in #18510 and #18315

It's well defined truncation here